File: app/src/main/java/com/airhud/app/data/HudModels.kt
package com.airhud.app.data

enum class HudPosition {
    TOP_RIGHT, TOP_LEFT, BOTTOM_RIGHT, BOTTOM_LEFT
}

enum class HudMode {
    HEAD_LOCKED, WORLD_LOCKED, SMOOTH_FOLLOW
}

enum class AiProvider {
    OPENAI, OPENROUTER, OLLAMA, CUSTOM
}

data class WidgetToggles(
    val clockEnabled: Boolean = true,
    val batteryEnabled: Boolean = true,
    val notificationEnabled: Boolean = true,
    val aiEnabled: Boolean = true
)

data class AiConfig(
    val provider: AiProvider = AiProvider.OPENAI,
    val baseUrl: String = "https://api.openai.com/v1",
    val apiKey: String = "",
    val model: String = "gpt-4o-mini"
)

data class HudSettings(
    val hudEnabled: Boolean = false,
    val position: HudPosition = HudPosition.TOP_RIGHT,
    val sizePercent: Int = 25,
    val opacityPercent: Int = 90,
    val mode: HudMode = HudMode.HEAD_LOCKED,
    val widgets: WidgetToggles = WidgetToggles(),
    val aiConfig: AiConfig = AiConfig()
)

data class NotificationInfo(
    val appName: String,
    val title: String,
    val text: String,
    val timestampMillis: Long
)

data class HudRuntimeState(
    val serviceRunning: Boolean = false,
    val clockText: String = "--:--",
    val dateText: String = "",
    val batteryPercent: Int = -1,
    val isCharging: Boolean = false,
    val notification: NotificationInfo? = null,
    val aiPrompt: String = "",
    val aiResponse: String? = null,
    val aiLoading: Boolean = false,
    val aiError: String? = null
)

File: unity/AirHUD-Spatial/Assets/Scripts/Data/HudModels.cs
using System;

namespace AirHud.Data
{
    public enum HudPosition { TopRight, TopLeft, BottomRight, BottomLeft }

    public enum HudMode { HeadLocked, WorldLocked, SmoothFollow }

    public enum AiProvider { OpenAI, OpenRouter, Ollama, Custom }

    [Serializable]
    public class WidgetToggles
    {
        public bool clockEnabled = true;
        public bool batteryEnabled = true;
        public bool notificationEnabled = true;
        public bool aiEnabled = true;
    }

    [Serializable]
    public class AiConfig
    {
        public AiProvider provider = AiProvider.OpenAI;
        public string baseUrl = "https://api.openai.com/v1";
        public string apiKey = "";
        public string model = "gpt-4o-mini";
    }

    [Serializable]
    public class HudSettings
    {
        public bool hudEnabled = false;
        public HudPosition position = HudPosition.TopRight;
        public int sizePercent = 25;
        public int opacityPercent = 90;
        public HudMode mode = HudMode.HeadLocked;
        public WidgetToggles widgets = new WidgetToggles();
        public AiConfig aiConfig = new AiConfig();
    }

    [Serializable]
    public class NotificationInfo
    {
        public string appName;
        public string title;
        public string text;
        public long timestampMillis;
    }

    public class HudRuntimeState
    {
        public bool serviceRunning;
        public string clockText = "--:--";
        public string dateText = "";
        public int batteryPercent = -1;
        public bool isCharging;
        public NotificationInfo notification;
        public string aiPrompt = "";
        public string aiResponse;
        public bool aiLoading;
        public string aiError;
    }
}
